src/AutoRest.CSharp.V3/Generation/Writers/ModelWriter.cs

```diff
@@ -195,15 +195,23 @@ private static void WriteConstructor(CodeWriter writer, ObjectType schema)
 
                 using (writer.Scope())
                 {
-                    writer.WriteParameterNullChecks(constructor.Parameters);
+                    foreach (var parameter in constructor.Parameters)
+                    {
+                        if (parameter.DefaultValue == null)
```

Comment:
Why do we need this condition?

Reply:
This condition is for these kind of cases - https://github.com/ShivangiReja/autorest.csharp/blob/Generate_Content.json/test/TestServerProjects/constants/Generated/Models/ModelAsStringRequiredOneValueDefault.cs#L17

if I will remove this condition it will go inside and the generated code would look like this:

```csharp
parameter = parameter ?? ModelAsStringRequiredOneValueDefaultEnum.Value1;

Parameter = parameter ?? ModelAsStringRequiredOneValueDefaultEnum.Value1;
```

Reply:
I think it's the same problem as https://github.com/Azure/autorest.csharp/pull/846#discussion_r461040400